modules/adnuntiusBidAdapter.js

```diff
@@ -171,25 +146,69 @@ const storageTool = (function () {
           return voidAuId.auId;
         });
       }
-      metaInternal.segments = getSegmentsFromOrtb(ortb2);
-      metaInternal.kv = getKvsFromOrtb(ortb2);
     },
     saveToStorage: function (serverData, network) {
       setMetaInternal(serverData, network);
     },
     getUrlRelatedData: function () {
       // getting the URL information is theoretically not network-specific
-      const { segments, kv, usi, voidAuIdsArray } = metaInternal;
-      return { segments, kv, usi, voidAuIdsArray };
+      const { usi, voidAuIdsArray } = metaInternal;
+      return { usi, voidAuIdsArray };
     },
     getPayloadRelatedData: function (network) {
       // getting the payload data should be network-specific
-      const { segments, kv, usi, userId, voidAuIdsArray, voidAuIds, ...payloadRelatedData } = getMetaDataFromLocalStorage(network).reduce((a, entry) => ({ ...a, [entry.key]: entry.value }), {});
+      const { segments, usi, userId, voidAuIdsArray, voidAuIds, ...payloadRelatedData } = getMetaDataFromLocalStorage(network).reduce((a, entry) => ({ ...a, [entry.key]: entry.value }), {});
       return payloadRelatedData;
     }
   };
 })();
 
+const targetingTool = (function() {
+  const getSegmentsFromOrtb = function(bidderRequest) {
+    const userData = deepAccess(bidderRequest.ortb2 || {}, 'user.data');
+    let segments = [];
+    if (userData) {
+      userData.forEach(userdat => {
+        if (userdat.segment) {
+          segments.push(...userdat.segment.map((segment) => {
+            if (isStr(segment)) return segment;
+            if (isStr(segment.id)) return segment.id;
+          }).filter((seg) => !!seg));
+        }
+      });
+    }
+    return segments
+  };
+
+  const getKvsFromOrtb = function(bidderRequest) {
+    return deepAccess(bidderRequest.ortb2 || {}, 'site.ext.data');
+  };
+
+  return {
+    addSegmentsToUrlData: function (validBids, bidderRequest, existingUrlRelatedData) {
+      let segments = getSegmentsFromOrtb(bidderRequest || {});
+
+      for (let i = 0; i < validBids.length; i++) {
+        const bid = validBids[i];
+        const targeting = bid.params.targeting || {};
```

Comment:
there seems to be a new way to send you segments through the bid.params.targeting but it is not documented in Prebid docs, is there a reason for that ?

Reply:
There is a general `targeting` option as described in the documentation: https://docs.prebid.org/dev-docs/bidders/adnuntius.html

That targeting option is quite flexible and can include many different things that our adserver supports, for instance: `kv`, `auml`, `keywords`, `segments` and, as the documentation demonstrates, `c`.

So it's not a new way to send through`segments` -- it's always been the case that segments could be passed through along with the other options our Adnuntius ad server supports.